en/deploy-br-federation-manager.md

```diff
@@ -0,0 +1,350 @@
+  namespace: tidb-admin
+```
+
+#### Generate Kubeconfig
+
+{{< copyable "shell-regular" >}}
+
+```shell
+export USER_TOKEN_NAME=$(kubectl -n tidb-admin get serviceaccount br-federation-member -o=jsonpath='{.secrets[0].name}')
+export USER_TOKEN_VALUE=$(kubectl -n tidb-admin get secret/${USER_TOKEN_NAME} -o=go-template='{{.data.token}}' | base64 --decode)
+export CURRENT_CONTEXT=$(kubectl config current-context)
+export CURRENT_CLUSTER=$(kubectl config view --raw -o=go-template='{{range .contexts}}{{if eq .name "'''${CURRENT_CONTEXT}'''"}}{{ index .context "cluster" }}{{end}}{{end}}')
+export CLUSTER_CA=$(kubectl config view --raw -o=go-template='{{range .clusters}}{{if eq .name "'''${CURRENT_CLUSTER}'''"}}"{{with index .cluster "certificate-authority-data" }}{{.}}{{end}}"{{ end }}{{ end }}')
+export CLUSTER_SERVER=$(kubectl config view --raw -o=go-template='{{range .clusters}}{{if eq .name "'''${CURRENT_CLUSTER}'''"}}{{ .cluster.server }}{{end}}{{ end }}')
+
+cat << EOF > {k8s-name}-kubeconfig
+apiVersion: v1
+kind: Config
+current-context: ${CURRENT_CONTEXT}
+contexts:
+- name: ${CURRENT_CONTEXT}
+  context:
+    cluster: ${CURRENT_CONTEXT}
+    user: br-federation-member
+    namespace: kube-system
+clusters:
+- name: ${CURRENT_CONTEXT}
+  cluster:
+    certificate-authority-data: ${CLUSTER_CA}
+    server: ${CLUSTER_SERVER}
+users:
+- name: br-federation-member
+  user:
+    token: ${USER_TOKEN_VALUE}
+EOF
+```
+
+### Control Plane
+
+You should select one Kubernetes cluster as control plane to deploy BR Federation Manager. You need only execute steps below in the control plane.
+
+#### Create CRD
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl create -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/manifests/federation-crd.yaml
+```
+
+#### Create Secret
+
+You already have kubeconfig files of data planes. Now, you need encode all the kubeconfig files to a secret.
+
+Firstly, encode every kubeconfig file by `base64 -i {k8s-name}-kubeconfig`.
+
+Secondly, put all the output of first step to a secret object.
+
+{{< copyable "shell-regular" >}}
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: br-federation-kubeconfig
+type: Opaque
+data:
+  {k8s-name1}: {encoded-kubeconfig-1}
+  {k8s-name2}: {encoded-kubeconfig-2}
+  ...
+```
+
+#### Install BR Federation Manager
+
+1. Create a namespace to create resources related to br federation manager.
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl create ns br-fed-admin
+```
+2. Create the secret that contains all the encoded kubeconfig files in the namespace
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl create -f {secret}.yaml -n br-fed-admin
+```
+3. Download tidb-operator repo and switch to release-1.5 branch. Enter the Helm chart directory `./charts/br-federation`
+and install the Helm chart.
+
+{{< copyable "shell-regular" >}}
+
+```shell
+helm install --namespace br-fed-admin br-federation-manager . --set image=${br-federation-image}
```

Comment:
wondering why we need to secret above for this? we got `MountVolume.SetUp failed for volume "federation-kubeconfig" : secret "br-federation-kubeconfig" not found` error during container creation.

Reply:
I think if we get the whole manifest it should mounting this secrete to an environment var.

Reply:
The name of the secret can be configured by the `values.yaml`, the default configuration is [here](https://github.com/pingcap/tidb-operator/blob/release-1.5/charts/br-federation/values.yaml#L24)